evalml/tests/component_tests/test_decision_tree_classifier.py

```diff
@@ -0,0 +1,64 @@
+import numpy as np
+from sklearn.tree import DecisionTreeClassifier as SKDecisionTreeClassifier
+
+from evalml.model_family import ModelFamily
+from evalml.pipelines import DecisionTreeClassifier
+from evalml.problem_types import ProblemTypes
+
+
+def test_model_family():
+    assert DecisionTreeClassifier.model_family == ModelFamily.DECISION_TREE
+
+
+def test_problem_types():
+    assert ProblemTypes.BINARY in DecisionTreeClassifier.supported_problem_types
+    assert ProblemTypes.MULTICLASS in DecisionTreeClassifier.supported_problem_types
+    assert len(DecisionTreeClassifier.supported_problem_types) == 2
+
+
+def test_fit_predict_binary(X_y_binary):
+    X, y = X_y_binary
+
+    sk_clf = SKDecisionTreeClassifier(max_depth=6, max_features='auto', random_state=0)
+    sk_clf.fit(X, y)
+    y_pred_sk = sk_clf.predict(X)
+    y_pred_proba_sk = sk_clf.predict_proba(X)
+
+    clf = DecisionTreeClassifier()
+    clf.fit(X, y)
+    y_pred = clf.predict(X)
+    y_pred_proba = clf.predict_proba(X)
+
+    np.testing.assert_almost_equal(y_pred, y_pred_sk, decimal=5)
+    np.testing.assert_almost_equal(y_pred_proba, y_pred_proba_sk, decimal=5)
+
+
+def test_fit_predict_multi(X_y_multi):
+    X, y = X_y_multi
+
+    sk_clf = SKDecisionTreeClassifier(max_depth=6, max_features='auto', random_state=0)
+    sk_clf.fit(X, y)
+    y_pred_sk = sk_clf.predict(X)
+    y_pred_proba_sk = sk_clf.predict_proba(X)
+
+    clf = DecisionTreeClassifier()
+    clf.fit(X, y)
+    y_pred = clf.predict(X)
+    y_pred_proba = clf.predict_proba(X)
+
+    np.testing.assert_almost_equal(y_pred, y_pred_sk, decimal=5)
+    np.testing.assert_almost_equal(y_pred_proba, y_pred_proba_sk, decimal=5)
+
+
+def test_feature_importance(X_y_binary):
+    X, y = X_y_binary
+
+    clf = DecisionTreeClassifier()
+    sk_clf = SKDecisionTreeClassifier(max_depth=6, max_features='auto', random_state=0)
+    sk_clf.fit(X, y)
+    sk_feature_importance = sk_clf.feature_importances_
+
+    clf.fit(X, y)
+    feature_importance = clf.feature_importance
+
+    np.testing.assert_almost_equal(sk_feature_importance, feature_importance, decimal=5)
```

Comment:
Would be good to add a test for categorical data, as SKLearn's DTC fails with this currently

Reply:
Hmm, thinking about this some more: I can see the flip side of adding a test for categorical data if the DecisionTree estimators *should* be able to handle categorical data (to check that it's actually handling those columns correctly), but am more unclear about the value of adding a test to check whether or not categorical data breaks? @bchen1116 @freddyaboulton Maybe you guys can convince me otherwise haha

Reply:
Yeah, I think it would be much more useful to add support for categorical data, and then add tests for it to make sure the behavior is as expected!

Reply:
Mhm, but given that DecisionTree estimators don't support categorical data, I don't think the tests are necessary / adding support for categorical data within the estimator is necessary?

Reply:
Yea I agree that adding that test would make more sense if these estimators could natively handle categorical features. The thought is that if users use these components outside the context of AutoML "they know what they're doing " and it's on them to encode columns that need encoding. In the context of AutoML, we take care of adding encoders for categorical features in `make_pipeline`. However, I think there is value in expanding the test coverage for `make_pipeline` - It is using an outdated list of estimators. It would be good to add coverage for Catboost, LightGBM (these should not need encoders because they handle categorical features natively), ExtraTrees, DecisionTrees, etc.

Reply:
Yup, I agree, `make_pipeline` and its testing should be updated. The testing for ` make_pipeline` is pretty sloppy right now (I'm likely to be the person to blame for that 😂) and should probably dynamically get the available estimators so it doesn't get outdated so quickly. Will file something to update the method if necessary and add cleaner testing. Thanks guys!